src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameNodeDefinitions.cs

```diff
@@ -0,0 +1,426 @@
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+            => visitor.Visit(this);
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+        {
+            if (index == 0)
+            {
+                return OpenBracket;
+            }
+
+            if (index == ChildCount - 1)
+            {
+                return CloseBracket;
+            }
+
+            return CommaTokens[index - 1];
+        }
+    }
+
+    /// <summary>
+    /// The type argument list for a method declaration. 
+    /// 
+    /// <code>
+    /// Ex: MyType.MyMethod[T, U, V](T t, U u, V v) 
+    ///                    ^-----------------------  "[" = Open Token 
+    ///                     ^------^   ------------  "T, U, V" = SeparatedStackFrameNodeList&lt;StackFrameTypeArgumentNode&gt;
+    ///                             ^--------------  "]" = Close Token
+    /// </code>
+    /// 
+    /// </summary>
+    internal sealed class StackFrameTypeArgumentList : StackFrameNode
+    {
+        public readonly StackFrameToken OpenToken;
+        public readonly EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameIdentifierNameNode> TypeArguments;
+        public readonly StackFrameToken CloseToken;
+
+        public StackFrameTypeArgumentList(
+            StackFrameToken openToken,
+            EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameIdentifierNameNode> typeArguments,
+            StackFrameToken closeToken)
+            : base(StackFrameKind.TypeArgument)
+        {
+            Debug.Assert(openToken.Kind is StackFrameKind.OpenBracketToken or StackFrameKind.LessThanToken);
+            Debug.Assert(typeArguments.Length > 0);
+            Debug.Assert(openToken.Kind == StackFrameKind.OpenBracketToken ? closeToken.Kind == StackFrameKind.CloseBracketToken : closeToken.Kind == StackFrameKind.GreaterThanToken);
+
+            OpenToken = openToken;
+            TypeArguments = typeArguments;
+            CloseToken = closeToken;
+        }
+
+        internal override int ChildCount => TypeArguments.NodesAndTokens.Length + 2;
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+            => visitor.Visit(this);
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+        {
+            if (index >= ChildCount)
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (index == 0)
+            {
+                return OpenToken;
+            }
+
+            if (index == ChildCount - 1)
+            {
+                return CloseToken;
+            }
+
+            // Includes both the nodes and separator tokens as children
+            return TypeArguments.NodesAndTokens[index - 1];
+        }
+    }
+
+    internal sealed class StackFrameParameterDeclarationNode : StackFrameDeclarationNode
+    {
+        public readonly StackFrameNameNode Type;
```

Comment:
what if the parameter is something like `(int[] foo`)?  add test.  also, test many of the other C# types that come up regularly.